### PR TITLE
Refactor visitor tests for BinaryElementwiseArithmetic ops

### PR DESCRIPTION
**Pull request by dkozykowski**
### Details:
- Refactor visitor tests for BinaryElementwiseArithmetic ops

### Tickets:
 - 58428

### CODE REVIEW

File: ngraph/test/visitors/op/binary_elementwise.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#pragma once
+
+#include <vector>
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+
+#include "ngraph/op/util/attr_types.hpp"
+#include "util/visitor.hpp"
+
+using namespace ngraph;
```
Comment (pelszkow) [inline ngraph/test/visitors/op/binary_elementwise.hpp:14]: Don't use `using namespace` in header in global namespace.

File: ngraph/test/visitors/op/binary_elementwise.hpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+template <typename T>
+class BinaryOperatorVisitor : public testing::Test
+{
+};
+
+class BinaryOperatorTypeName
+{
+public:
+    template <typename T>
+    static std::string GetName(int)
```
Comment (pelszkow) [inline ngraph/test/visitors/op/binary_elementwise.hpp:32]: On the first glance this function should be in `BinaryOperator` (`BinaryOperatorType`).

File: ngraph/test/visitors/op/binary_elementwise.hpp (hunk 1 above)
Comment (dkozykowski) [reply ngraph/test/visitors/op/binary_elementwise.hpp]: Fixed, thank you!

File: ngraph/test/visitors/op/binary_elementwise.hpp (hunk 2 above)
Comment (dkozykowski) [reply ngraph/test/visitors/op/binary_elementwise.hpp]: It seems like we need a separate class for this function to pass it as an argument calling `INSTANTIATE_TYPED_TEST_SUITE_P`.

Comment (pelszkow) [reply ngraph/test/visitors/op/binary_elementwise.hpp]: Apparently yes.

File: ngraph/test/visitors/op/binary_ops.hpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+public:
+    template <typename T>
+    static std::string GetName(int)
+    {
+        using OP_Type = typename T::op_type;
+        constexpr ngraph::element::Type precision(T::element_type);
+        const ngraph::Node::type_info_t typeinfo = OP_Type::get_type_info_static();
+        std::string op_name{typeinfo.name};
+        op_name.append("_");
+        return (op_name.append(precision.get_type_name()));
```
Comment (pelszkow) [inline ngraph/test/visitors/op/binary_ops.hpp:34]: I possible that bellow example will use RVO.
```suggestion
        return std::string{typeinfo.name} + "_" + precision.get_type_name();
```

File: ngraph/test/visitors/op/binary_ops.hpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    ngraph::test::NodeBuilder::get_ops().register_factory<OP_Type>();
+    const auto A =
+        std::make_shared<ngraph::op::Parameter>(element_type, ngraph::PartialShape{1, 2, 3});
+    const auto B =
+        std::make_shared<ngraph::op::Parameter>(element_type, ngraph::PartialShape{3, 2, 1});
+
+    auto auto_broadcast = ngraph::op::AutoBroadcastType::NUMPY;
+
+    const auto op_func = std::make_shared<OP_Type>(A, B, auto_broadcast);
+    ngraph::test::NodeBuilder builder(op_func);
```
Comment (pelszkow) [inline ngraph/test/visitors/op/binary_ops.hpp:54]: Please add validation for attributes number. 
e.g.: https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/visitors/op/adaptive_max_pool.cpp#L26-L27

File: ngraph/test/visitors/op/add.cpp [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+#include "binary_ops.hpp"
+#include "ngraph/opsets/opset1.hpp"
+
+using Type = ::testing::Types<BinaryOperatorType<ngraph::opset1::Add, ngraph::element::f32>>;
+
+INSTANTIATE_TYPED_TEST_SUITE_P(visitor_with_auto_broadcast,
+                               BinaryOperatorVisitor,
+                               Type,
+                               BinaryOperatorTypeName);
```
Comment (pelszkow) [inline ngraph/test/visitors/op/add.cpp:13]: Add empty line on end of each file.
```suggestion
                               BinaryOperatorTypeName);

```

**Review by GlebKazantaev — APPROVED** [review]
Approve changes in ngraph operations.

File: ngraph/test/visitors/op/binary_ops.hpp (hunk 4 above)
Comment (dkozykowski) [reply ngraph/test/visitors/op/binary_ops.hpp:52]: Done!

File: ngraph/test/visitors/op/binary_ops.hpp (hunk 3 above)
Comment (dkozykowski) [reply ngraph/test/visitors/op/binary_ops.hpp]: Refactored just as you suggested. Thank you!

**Review by pelszkow — APPROVED** [review]
LGTM

File: ngraph/core/src/op/floor_mod.cpp [hunk 6]
```diff
@@ -10,7 +10,7 @@
 using namespace std;
 using namespace ngraph;
 
-constexpr NodeTypeInfo op::v1::FloorMod::type_info;
+NGRAPH_RTTI_DEFINITION(op::v1::FloorMod, "FloorMod", 1);
```
Comment (ilyachur) [inline ngraph/core/src/op/floor_mod.cpp:13]:
```suggestion
NGRAPH_RTTI_DEFINITION(op::v1::FloorMod, "FloorMod", 1, BinaryElementwiseArithmetic);
```

File: ngraph/core/src/op/broadcast.cpp [hunk 7]
```diff
@@ -17,7 +17,7 @@
 using namespace std;
 using namespace ngraph;
 
-constexpr NodeTypeInfo op::v3::Broadcast::type_info;
+NGRAPH_RTTI_DEFINITION(op::v3::Broadcast, "Broadcast", 3);
```
Comment (ilyachur) [inline ngraph/core/src/op/broadcast.cpp:20]:
```suggestion
NGRAPH_RTTI_DEFINITION(op::v3::Broadcast, "Broadcast", 3, BroadcastBase);
```

File: ngraph/core/src/op/maximum.cpp [hunk 8]
```diff
@@ -58,7 +58,7 @@ namespace maximumop
 
 // ------------------------------------ v1 -------------------------------------
 
-constexpr NodeTypeInfo op::v1::Maximum::type_info;
+NGRAPH_RTTI_DEFINITION(op::v1::Maximum, "Maximum", 1);
```
Comment (ilyachur) [inline ngraph/core/src/op/maximum.cpp:61]:
```suggestion
NGRAPH_RTTI_DEFINITION(op::v1::Maximum, "Maximum", 1, BinaryElementwiseArithmetic);
```
And for other operations too

File: ngraph/core/src/op/bucketize.cpp [hunk 9]
```diff
@@ -8,7 +8,7 @@
 using namespace ngraph;
 using namespace std;
 
-constexpr NodeTypeInfo op::v3::Bucketize::type_info;
+NGRAPH_RTTI_DEFINITION(op::v3::Bucketize, "Bucketize", 3);
```
Comment (dkozykowski) [inline ngraph/core/src/op/bucketize.cpp:12]: @ilyachur should 4th parameter be provided and what should it be, if operator inherits from Op class, like in example below?
`class NGRAPH_API Bucketize : public Op`

File: ngraph/core/src/op/maximum.cpp (hunk 8 above)
Comment (dkozykowski) [reply ngraph/core/src/op/maximum.cpp]: Done

**Review by sdurawa — APPROVED** [review]
LGTM, please merge master branch changes.

File: ngraph/core/src/op/bucketize.cpp (hunk 9 above)
Comment (ilyachur) [reply ngraph/core/src/op/bucketize.cpp:12]: No we shouldn't. Op class doesn't have RTTI declaration

**Review by mitruska — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**